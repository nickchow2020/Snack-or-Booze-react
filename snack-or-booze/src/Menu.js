import React from "react";
import { Link} from "react-router-dom";
import { usePathname } from "./Hooks";
import "./FoodMenu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

function Menu({items}) {

  const path = usePathname()

  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            Menu
          </CardTitle>
          <CardText>
            Some quick example text to build on the card title and make up the
            bulk of the card's content.
          </CardText>
          <ListGroup>
            {items.map(item => (
              <Link to={`${path}/${item.id}`} key={item.id}>
                <ListGroupItem>{item.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
        <Link to="/snacks/add" className="addNewLink">Add New Snack</Link>
      </Card>
    </section>
  );
}

export default Menu;
