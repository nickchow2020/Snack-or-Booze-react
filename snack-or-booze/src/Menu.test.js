import React from "react";
import {render,fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import {MemoryRouter} from "react-router-dom";
import Menu from "./Menu";


const snacks = [
    {
        "id": "nachos",
        "name": "Nachos",
        "description": "An American classic!",
        "recipe": "Cover expensive, organic tortilla chips with Cheez Whiz.",
        "serve": "Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
    },
    {
        "id": "hummus",
        "name": "Hummus",
        "description": "Sure to impress your vegan friends!",
        "recipe": "Purchase one container of hummus.",
        "serve": "Place unceremoniously on the table, along with pita bread."
    },
    {
        "id": "arugula-and-walnut-salad",
        "name": "Arugula and Walnut Salad",
        "description": "Tart and delicious.",
        "recipe": "Mix arugula, toasted walnuts, and thinly-sliced Parmesan cheese. Dress with lemon and olive oil.",
        "serve": "Place on tiny, precious little plates."
    }
]

it("should it render",function(){
    render(
    <MemoryRouter>
        <Menu items={snacks} />
    </MemoryRouter>
    )
})

it("should it match snapshot",function(){
    const {asFragment} = render(
        <MemoryRouter>
            <Menu items={snacks} />
        </MemoryRouter>
        )

    expect(asFragment()).toMatchSnapshot()
})


it("should it match context",function(){
    const {getByText} = render(
        <MemoryRouter initialEntries={["/snacks"]}>
            <Menu items={snacks} />
        </MemoryRouter>
        )

    expect(getByText("Nachos")).toBeInTheDocument()
})

it("should it show form",function(){
    const {getByText} = render(
        <MemoryRouter initialEntries={["/snacks"]}>
            <Menu items={snacks} />
        </MemoryRouter>
        )
    
    const addNewFormBtn = getByText("Add New Snack")
    fireEvent.click(addNewFormBtn)

    expect(getByText("Add New Snack")).toBeInTheDocument()
})



