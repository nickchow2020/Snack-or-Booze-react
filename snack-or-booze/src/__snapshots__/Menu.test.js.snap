// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should it match snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Menu
        </h5>
        <p
          class="card-text"
        >
          Some quick example text to build on the card title and make up the bulk of the card's content.
        </p>
        <ul
          class="list-group"
        >
          <a
            href="//nachos"
          >
            <li
              class="list-group-item"
            >
              Nachos
            </li>
          </a>
          <a
            href="//hummus"
          >
            <li
              class="list-group-item"
            >
              Hummus
            </li>
          </a>
          <a
            href="//arugula-and-walnut-salad"
          >
            <li
              class="list-group-item"
            >
              Arugula and Walnut Salad
            </li>
          </a>
        </ul>
      </div>
      <a
        class="addNewLink"
        href="/snacks/add"
      >
        Add New Snack
      </a>
    </div>
  </section>
</DocumentFragment>
`;
